parachain/pallets/ethereum-beacon-client/src/lib.rs

```diff
@@ -385,7 +390,7 @@ pub mod pallet {
 			let sync_committee = Self::get_sync_committee_for_period(current_period)?;
 
 			let body_root = merkleization::hash_tree_root_beacon_body(update.block.body.clone())
-				.map_err(|_| DispatchError::Other("Beacon body hash tree root failed"))?;
+				.map_err(|_| Error::<T>::HashTreeRootFailed)?;
```

Comment:
Is it OK to use the same `HashTreeRootFailed` error for all these different hash tree root failures?

Reply:
It could make differenr hash tree root failures harder to debug.

Reply:
Good point, added an error per struct in https://github.com/Snowfork/snowbridge/pull/696/commits/5470b333f3b089bf8eaa8a8b9d235a24341dad55.